implementation/src/main/java/io/smallrye/config/ConfigSourceInterceptorContext.java

```diff
@@ -14,4 +14,6 @@
      * @return a {@link ConfigValue} with information about the key, lookup value and source ConfigSource.
      */
     ConfigValue proceed(String name);
+
+    SecretKeys getSecretKeys();
```

Comment:
I think adding special purpose features like this one on to general purpose interfaces risks a large growth.

From the end user perspective, IMO ideally they should only "see" the API concept of secret keys when the config source is being built, and when (globally) turning on and off secrets.  The config source builder can build an interceptor which has the key set right on it, so the intermediate abstraction of `SecretKeys` should not be needed.

Reply:
I've added this here so an interceptor could find is a particular key is a secret, in the case of a logging interceptor, to exclude it from logging.

For instance, when you open up the secrets to perform injection, it would open to the entire interceptor chain, so a logging interceptor has access to the secret being injected and print the result.  We would need to disable access in a middle of an enable access and I think it might be a little confusing.

We can certainly pass in the Set to the logging interceptor as well, and make the check as we would do in the security one and skip the log. On the other hand, I think it is useful to have some kind of API that tells you is a certain key `isSecret`. Or we can just expose the Set directly.

Reply:
TBH I imagined loggers just catching the exception.  Otherwise there's yet another piece of information that has to be propagated along with the configuration value.

Reply:
Ah no, it is the logger to log the config name, source and value, so you know what was loaded. It seems to be a common request, at least seen some people asking about it. In that case, you need to lock the keys again in the case you run in an unlocked context so you don't print the value accidentally.

Reply:
Yes, that is understood.  I mean, I expect that things which may log locked properties will surround accesses with try/catch, and skip entries for which an exception was thrown.

Reply:
Yes, but in the case where the lock is unlocked before the chain, you still want the logger to be locked.

Reply:
So, the general purpose logger, should always run its code in a new lock.